app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/BankAccountController.scala

```diff
@@ -257,7 +257,14 @@ object BankAccountController {
   val accountNumberRegex: Predicate[String]        = "^\\d{6,8}$".r.pattern.asPredicate()
   val accountNumberMapping: Mapping[AccountNumber] =
     nonEmptyText(minLength = 6, maxLength = 8)
-      .transform[AccountNumber](s => AccountNumber(s.replaceAllLiterally(" ", "")), _.value)
+      .transform[AccountNumber](
+        s => {
+          val cleanNumber  = s.replaceAll("""[\s-]""", "")
+          val paddedNumber = ("00" + cleanNumber).substring(2 + cleanNumber.length - 8)
+          AccountNumber(paddedNumber)
```

Comment:
Why did you decided to go with Regex? People tend to say you "you had 1 problem and then you decided to use  Regex and got 2 problems now". Although it is joke but there is some true about it.
Why don't use already an existing solution:
https://stackoverflow.com/questions/8131291/how-to-convert-an-int-to-a-string-of-a-given-length-with-leading-zeros-to-align
It seems to be battle tested and proven to work.

Reply:
Line 262, `s` is a String, first I need to remove spaces (that requires the regex), then if I go your way I need to parse it to an Int, and then turn it back to a string again. That may throw an exception ... that's more problems, and I did not get rid of the regex.